src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -611,10 +611,10 @@ export class FlexibleConnectedPositionStrategy implements PositionStrategy {
 
     // Determine how much the overlay goes outside the viewport on each
     // side, which we'll use to decide which direction to push it.
-    const overflowRight = Math.max(start.x + overlay.width - viewport.width, 0);
-    const overflowBottom = Math.max(start.y + overlay.height - viewport.height, 0);
-    const overflowTop = Math.max(viewport.top - scrollPosition.top - start.y, 0);
-    const overflowLeft = Math.max(viewport.left - scrollPosition.left - start.x, 0);
+    const overflowRight = Math.max(start.x + overlay.width - viewport.width - this._viewportMargin, 0);
+    const overflowBottom = Math.max(start.y + overlay.height - viewport.height - this._viewportMargin, 0);
+    const overflowTop = Math.max(viewport.top - start.y, 0);
+    const overflowLeft = Math.max(viewport.left - start.x, 0);
```

Comment:
I'm not sure I follow the changes for `top` and `left`. Also if you take a look at `_getNarrowedViewportRect` which generates the `_viewportRect`, it's already removing the viewport margin for each side so removing it again from `right` and `bottom` might be too much.

Reply:
This is the same as `start.x + overlay.width - (viewport.width + this._viewportMargin)`.

Would it be more understandable to be written in that way?

Reply:
I changed it in that way.

Reply:
Oh sorry my mistake. The () is not needed.